server/src/main/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimGroupProvisioning.java

```diff
@@ -222,10 +231,32 @@ public ScimGroup retrieve(String id, final String zoneId) throws ScimResourceNot
         }
     }
 
+    @SuppressWarnings("java:S1874")
+    private Set<String> getAllowedUserGroups(String zoneId) {
+        if (zoneId == null) {
+            throw new IllegalArgumentException("zoneId is null");
```

Comment:
this here is not tested, but if you change 
 (zoneId.equals(currentZone.getId())) 
to
 (currentZone.getId(equals(zoneId))

you dont need the null check for zoneId

Reply:
done with [ab2d5cf](https://github.com/cloudfoundry/uaa/pull/2606/commits/ab2d5cf74037ce00effc502395c8b2578be87c2e)
(line 241) "jdbcIdentityZoneProvisioning.retrieve(zoneId)" seems to be null-safe as well